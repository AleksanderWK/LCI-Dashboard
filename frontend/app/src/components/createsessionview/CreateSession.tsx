import React, {useEffect, useState} from "react";
import {
    makeStyles,
    createStyles,
    Theme,
    Typography,
    TextField,
    Button,
    Input,
    Select,
    MenuItem,
    InputLabel,
    FormControl,
    IconButton,
    SvgIconProps,
    SvgIcon
} from "@material-ui/core";
import AddCircleRoundedIcon from "@material-ui/icons/AddCircleRounded";
import {useRecoilCallback, useRecoilState, useResetRecoilState, useSetRecoilState} from "recoil";
import {ipcGet, ipcInvoke, ipcOnce, ipcSend} from "../../ipc";
import {createSessionValuesState} from "../../state/createSession";
import {addStudentPopupOpenState, createSessionPopupOpenState} from "../../state/popup";
import {Student, studentsState} from "../../state/student";
import {useHistory, useLocation} from "react-router-dom";
import {
    selectedSessionIdState,
    Session,
    sessionIdsState,
    sessionRecordingState,
    sessionState
} from "../../state/session";
import {EyeTrackingDevice} from "../../constants";
import {StyledTooltipBottom, StyledTooltipRight} from "../common/Tooltips";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        form: {
            display: "grid",
            gridTemplateColumns: "1fr",
            gap: 36,
            color: theme.palette.text.default
        },
        inputNewSessionName: {
            width: "100%"
        },
        hasText: {
            borderBottomColor: theme.palette.secondary.main
        },
        inputSelectStudent: {
            width: "250px"
        },
        addStudentBtn: {
            marginTop: "10px"
        },
        addStudentContainer: {
            display: "flex",
            justifyContent: "space-between"
        },
        sessionToken: {
            fontSize: "20px",
            fontWeight: 600,
            letterSpacing: "2px"
        },
        sessionInfoContainer: {
            height: "70px",
            display: "flex",
            flexDirection: "column",
            justifyContent: "space-between"
        },
        btn: {
            color: theme.palette.text.default
        },
        placeholder: {
            color: theme.palette.text.placeholder
        },
        noStudentsPromptContainer: {
            width: "250px",
            height: "auto",
            color: theme.palette.text.placeholder
        }
    })
);

function CopyIcon(props: SvgIconProps): JSX.Element {
    return (
        <SvgIcon {...props}>
            <path
                _ngcontent-axr-c9=""
                clipRule="evenodd"
                d="M16.4998 14.1666V2.49992C16.4998 1.58325 15.7498 0.833252 14.8332 0.833252H5.6665C4.74984 0.833252 3.99984 1.58325 3.99984 2.49992V14.1666C3.99984 15.0833 4.74984 15.8333 5.6665 15.8333H14.8332C15.7498 15.8333 16.4998 15.0833 16.4998 14.1666ZM13.9998 17.4999H2.33317V5.83325H0.666504V17.4999C0.666504 18.4166 1.4165 19.1666 2.33317 19.1666H13.9998V17.4999ZM5.6665 14.1666H14.8332V2.49992H5.6665V14.1666Z"
                fillRule="evenodd"
            ></path>
        </SvgIcon>
    );
}

export default function CreateSession(): JSX.Element {
    const classes = useStyles();

    const [createSessionValues, setCreateSessionValues] = useRecoilState(createSessionValuesState);
    const resetCreateSessionValues = useResetRecoilState(createSessionValuesState);

    const [sessionNameNotSet, setSessionNameNotSet] = useState(true);
    const [studentNameNotSet, setStudentNameNotSet] = useState(true);

    const setAddStudentPopupOpen = useSetRecoilState(addStudentPopupOpenState);
    const setCreateSessionPopupOpen = useSetRecoilState(createSessionPopupOpenState);

    const [students, setStudents] = useRecoilState(studentsState);

    const history = useHistory();
    const location = useLocation();

    useEffect(() => {
        // Get users from DB when component loads
        ipcGet<Student[]>("getUsers").then((students) => {
            setStudents(students);
        });

        ipcOnce("readyConnection", (event: any, data: any) => {
            setCreateSessionValues((prevValues) => ({
                ...prevValues,
                studentConnected: true
            }));
        });
    }, []);

    const handleSelectionChange = (selection: string, value: string) => {
        let student = undefined;

        if (selection == "studentId") {
            student = students.find((student) => student._id == value);
        }

        // Overwrite an attribute based on the selection parameter
        const updatedSessionSelections = {
            ...createSessionValues,
            [selection]: value,
            ...(selection == "studentId" && student && {studentName: student.name})
        };

        // Save updated menu values to state
        setCreateSessionValues(updatedSessionSelections);
    };

    const createSession = useRecoilCallback(({set}) => (session: Session) => {
        set(sessionState(session._id), session);

        set(sessionIdsState, (prevValue) => [...prevValue, session._id]);

        set(sessionRecordingState(session._id), {status: false, startTime: null, recordingId: null});

        set(selectedSessionIdState, session._id);
    });

    const handleCreateSession = () => {
        ipcSend("startDatastream", createSessionValues.sessionCode);

        const session: Partial<Session> = {
            sessionName: createSessionValues.sessionName,
            studentId: createSessionValues.studentId,
            studentName: createSessionValues.studentName,
            eyeTrackingDevice:
                createSessionValues.eyeTracker === "Mobile" ? EyeTrackingDevice.Mobile : EyeTrackingDevice.Stationary,
            startTime: new Date().getTime(),
            endTime: null,
            sessionCode: createSessionValues.sessionCode
        };

        ipcInvoke("insertSession", session).then((sessionId) => {
            session._id = sessionId as number;
            createSession(session as Session);
        });

        // Reset values
        resetCreateSessionValues();

        if (location.pathname === "/create-session") {
            history.push("session");
        } else if (location.pathname === "/session") {
            setCreateSessionPopupOpen(false);
        }
    };

    useEffect(() => {
        if (createSessionValues.sessionName) {
            setSessionNameNotSet(false);
        } else {
            setSessionNameNotSet(true);
        }

        if (createSessionValues.studentId) {
            setStudentNameNotSet(false);
        } else {
            setStudentNameNotSet(true);
        }

        if (!createSessionValues.sessionCode) {
            ipcGet("getCode").then((code: any) => {
                handleSelectionChange("sessionCode", code);
            });
        }
    }, [createSessionValues]);

    return (
        <div className={classes.form}>
            <Typography variant="h1" style={{color: "#000000"}}>
                Create New Learning Session
            </Typography>

            <TextField
                className={classes.inputNewSessionName}
                label={"Session name"}
                onChange={(event: React.ChangeEvent<{value: unknown}>) => {
                    handleSelectionChange("sessionName", event.target.value as string);
                }}
                value={createSessionValues.sessionName}
            />

            <div className={classes.addStudentContainer}>
                <FormControl>
                    <InputLabel id="label" className={classes.placeholder}>
                        Student
                    </InputLabel>
                    <Select
                        labelId="label"
                        onClose={() => {
                            // Remove input field focus when selection is closed
                            setTimeout(() => {
                                (document.activeElement as HTMLElement).blur();
                            }, 60);
                        }}
                        input={<Input />}
                        classes={{root: classes.inputSelectStudent}}
                        onChange={(event: React.ChangeEvent<{value: unknown}>) => {
                            handleSelectionChange("studentId", event.target.value as string);
                        }}
                        value={
                            students.some((student) => student._id === createSessionValues.studentId)
                                ? createSessionValues.studentId
                                : ""
                        }
                    >
                        {students.length > 0 ? (
                            students.map((student: Student) => (
                                <MenuItem key={student._id} value={student._id} data-testid={student.name}>
                                    {student.name}
                                </MenuItem>
                            ))
                        ) : (
                            <MenuItem
                                className={classes.noStudentsPromptContainer}
                                disabled
                                style={{whiteSpace: "normal", margin: "auto", textAlign: "center"}}
                            >
                                <p>You have not added any students yet. Click on the + button to add a new student.</p>
                            </MenuItem>
                        )}
                    </Select>
                </FormControl>
                <StyledTooltipBottom title="Add student" placement="bottom">
                    <IconButton
                        color="primary"
                        aria-label="add new student"
                        className={classes.addStudentBtn}
                        onClick={() => {
                            setAddStudentPopupOpen(true);
                        }}
                    >
                        <AddCircleRoundedIcon />
                    </IconButton>
                </StyledTooltipBottom>
            </div>
            {
                // TODO: Implement constraints in frontend based on the chosen eye tracking device,
                // i.e. which variables the user can choose to view.
                /*
                <div>
                    <Typography variant="caption">Eye tracking device</Typography>
                    <RadioGroup
                        row
                        aria-label="eye tracker decive"
                        name="eyetracker"
                        onChange={(event: React.ChangeEvent<{value: unknown}>) => {
                            handleSelectionChange("eyeTracker", event.target.value as string);

                            setDeviceNotSet(!event.target.value);
                        }}
                        value={createSessionValues.eyeTracker}
                    >
                        <FormControlLabel value="stationary" control={<Radio />} label="Stationary" />
                        <FormControlLabel value="mobile" control={<Radio />} label="Mobile" style={{marginLeft: "20px"}} />
                    </RadioGroup>
                </div>
                */
            }

            <div className={classes.sessionInfoContainer}>
                <Typography variant="caption">Session code</Typography>
                <div>
                    <Typography variant="caption" className={classes.sessionToken}>
                        {createSessionValues.sessionCode}
                    </Typography>
                    <StyledTooltipRight title="Copy" placement="right">
                        <IconButton
                            aria-label="copy-to-clipboard"
                            size="small"
                            style={{margin: "0px 0px 7px 5px"}}
                            onClick={() => {
                                navigator.clipboard.writeText(createSessionValues.sessionCode);
                            }}
                        >
                            <CopyIcon viewBox="-3 -3 25 25" />
                        </IconButton>
                    </StyledTooltipRight>
                </div>
                {createSessionValues.studentConnected ? (
                    <Typography variant="caption" style={{color: "#5BA350"}}>
                        Student has connected
                    </Typography>
                ) : (
                    <Typography variant="caption" style={{color: "#DD5757"}}>
                        Waiting for student connection...
                    </Typography>
                )}
            </div>
            <div>
                <Button
                    variant="contained"
                    color="primary"
                    onClick={handleCreateSession}
                    disabled={sessionNameNotSet || studentNameNotSet || !createSessionValues.studentConnected}
                    name="submitBtn"
                    className={classes.btn}
                    style={{marginRight: 8}}
                >
                    Create session
                </Button>
                <Button
                    onClick={() => {
                        if (location.pathname === "/create-session") {
                            history.push("/");
                            resetCreateSessionValues();
                        } else if (location.pathname === "/session") {
                            setCreateSessionPopupOpen(false);
                            resetCreateSessionValues();
                        }
                    }}
                    name="cancelBtn"
                    className={classes.btn}
                >
                    Cancel
                </Button>
            </div>
        </div>
    );
}
