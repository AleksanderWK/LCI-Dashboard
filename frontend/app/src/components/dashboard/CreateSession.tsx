import react, {useState} from "react";
import {
    makeStyles,
    createStyles,
    Theme,
    Typography,
    TextField,
    RadioGroup,
    FormControlLabel,
    Radio,
    Button,
    Input,
    Select,
    MenuItem,
    InputLabel,
    FormControl,
    IconButton
} from "@material-ui/core";
import AddCircleRoundedIcon from "@material-ui/icons/AddCircleRounded";
import React from "react";

const useStyles = makeStyles((theme: Theme) =>
    createStyles({
        paperStyles: {
            width: "325px",
            height: "450px",
            display: "flex",
            flexDirection: "column",
            justifyContent: "space-between",
            color: theme.palette.text.default
        },
        inputNewSessionName: {
            width: "100%"
        },
        hasText: {
            borderBottomColor: theme.palette.secondary.main
        },
        inputSelectStudent: {
            width: "250px"
        },
        addStudentBtn: {
            marginTop: "10px"
        },
        addStudentContainer: {
            display: "flex",
            justifyContent: "space-between"
        },
        sessionToken: {
            fontSize: "20px",
            fontWeight: 600,
            letterSpacing: "2px"
        },
        sessionInfoContainer: {
            height: "70px",
            display: "flex",
            flexDirection: "column",
            justifyContent: "space-between"
        },
        btn: {
            color: theme.palette.text.default,
            fontWeight: "bold"
        },
        placeholder: {
            color: theme.palette.text.placeholder
        }
    })
);

export interface sessionSelections {
    sessionName: string;
    studentName: string;
    eyeTracker: string;
}

interface Props {
    handleTogglePopup: (open: boolean) => void;
}

export default function CreateSession(props: Props): JSX.Element {
    const classes = useStyles();

    //addStudentPopup will save previous student names, create session will fetch these names
    const students: string[] = ["John Doe", "Jane Smith"];

    const [sessionSelections, setSessionSelections] = useState<sessionSelections>();
    const [studentConnected, setStudentConnected] = useState<boolean>(false);
    const [sessionToken, setSessionToken] = useState<number>();

    const handleSelectionChange = (selection: string, value: string) => {
        // Overwrite an attribute based on the selection parameter
        const updatedSessionSelections = {
            ...sessionSelections!,
            [selection]: value
        };

        // Save updated menu values to state
        setSessionSelections(updatedSessionSelections);
    };

    const togglePopup = () => {
        //toggles popup
    };

    const handelCreateSession = () => {
        console.log("create session");
    };

    return (
        <div className={classes.paperStyles}>
            <Typography variant="h1" style={{color: "#000000"}}>
                Create New Learning Session
            </Typography>
            <TextField
                className={classes.inputNewSessionName}
                label={"Session Name"}
                onChange={(event: React.ChangeEvent<{value: unknown}>) =>
                    handleSelectionChange("sessionName", event.target.value as string)
                }
            />
            <div className={classes.addStudentContainer}>
                <FormControl>
                    <InputLabel id="label" className={classes.placeholder}>
                        Student
                    </InputLabel>
                    <Select
                        labelId="label"
                        onClose={() => {
                            // Remove input field focus when selection is closed
                            setTimeout(() => {
                                (document.activeElement! as HTMLElement).blur();
                            }, 60);
                        }}
                        input={<Input />}
                        classes={{root: classes.inputSelectStudent}}
                        onChange={(event: React.ChangeEvent<{value: unknown}>) =>
                            handleSelectionChange("studentName", event.target.value as string)
                        }
                    >
                        {students.map((option) => (
                            <MenuItem key={option as string} value={option as string}>
                                {option}
                            </MenuItem>
                        ))}
                    </Select>
                </FormControl>
                <IconButton
                    color="primary"
                    aria-label="add new student"
                    className={classes.addStudentBtn}
                    onClick={togglePopup}
                >
                    <AddCircleRoundedIcon />
                </IconButton>
            </div>
            <div>
                <Typography variant="caption">Eye tracking device</Typography>
                <RadioGroup
                    row
                    aria-label="eye tracker decive"
                    name="eyetracker"
                    onChange={(event: React.ChangeEvent<{value: unknown}>) =>
                        handleSelectionChange("eyeTracker", event.target.value as string)
                    }
                >
                    <FormControlLabel value="stationary" control={<Radio />} label="Stationary" />
                    <FormControlLabel value="mobile" control={<Radio />} label="Mobile" style={{marginLeft: "20px"}} />
                </RadioGroup>
            </div>
            <div className={classes.sessionInfoContainer}>
                <Typography variant="caption">Session code</Typography>

                <Typography variant="caption" className={classes.sessionToken}>
                    748305
                </Typography>
                {studentConnected ? (
                    <Typography variant="caption" style={{color: "#5BA350"}}>
                        Student has connected
                    </Typography>
                ) : (
                    <Typography variant="caption" style={{color: "#DD5757"}}>
                        Waiting for student connection...
                    </Typography>
                )}
            </div>
            <Button
                variant="contained"
                color="primary"
                onClick={handelCreateSession}
                disabled={!studentConnected}
                className={classes.btn}
            >
                Create session
            </Button>
        </div>
    );
}
