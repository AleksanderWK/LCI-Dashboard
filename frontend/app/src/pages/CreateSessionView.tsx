import React, {Suspense} from "react";
import {makeStyles, createStyles} from "@material-ui/core";
import logo from "../assets/Images/LCI_logo.png";
import dashboardIllustration from "../assets/Images/dashboardIllustration.svg";
import CreateSession from "../components/createsessionview/CreateSession";
import PopupContainer from "../components/common/PopupContainer";
import AddStudent from "../components/createsessionview/AddStudent";
import {useRecoilState} from "recoil";
import {addStudentPopupOpenState} from "../state/popup";
import Popup from "../components/common/Popup";

const useStyles = makeStyles(() =>
    createStyles({
        pageContainer: {
            width: "100%",
            height: "100%",
            display: "grid",
            gridTemplateColumns: "30px 140px auto 140px 30px",
            gridTemplateRows: "30px auto 30px"
        },
        pageLogo: {
            gridColumnStart: 2,
            gridColumnEnd: 3,
            gridRowStart: 2,
            gridRowEnd: 3
        },
        pageContent: {
            display: "flex",
            justifyContent: "center",
            alignItems: "center",
            width: "100%",
            height: "100%",
            gridColumnStart: 3,
            gridColumnEnd: 4,
            gridRowStart: 2,
            gridRowEnd: 3
        },
        illustration: {
            width: "400px",
            height: "400px",
            marginLeft: "100px"
        }
    })
);

export default function CreateSessionView(): JSX.Element {
    const classes = useStyles();

    const [addStudentPopupOpen, setAddStudentPopupOpen] = useRecoilState(addStudentPopupOpenState);

    return (
        <>
            <div className={classes.pageContainer}>
                <img className={classes.pageLogo} src={logo} alt="The LCI-lab logo" />
                <div className={classes.pageContent}>
                    <Suspense fallback={<div>Loading...</div>}>
                        <CreateSession />
                    </Suspense>
                    <img className={classes.illustration} src={dashboardIllustration} alt="dashboard illustration" />
                </div>
            </div>

            <PopupContainer
                open={addStudentPopupOpen}
                onClose={() => {
                    setAddStudentPopupOpen(false);
                }}
            >
                <Popup>
                    <AddStudent />
                </Popup>
            </PopupContainer>
        </>
    );
}
