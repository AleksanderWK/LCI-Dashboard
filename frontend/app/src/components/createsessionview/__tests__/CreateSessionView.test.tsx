import React from "react";
import ReactDOM from "react-dom";
import {fireEvent, render, screen, waitFor} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import CreateSessionView from "../../../pages/CreateSessionView";
import CreateSession from "../CreateSession";

it("create session view matches snapshot", () => {
    const {baseElement} = render(<CreateSessionView />);
    expect(baseElement).toMatchSnapshot();
});

it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<CreateSessionView />, div);
});
