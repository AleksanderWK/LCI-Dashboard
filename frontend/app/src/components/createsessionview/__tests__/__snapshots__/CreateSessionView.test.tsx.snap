// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create session view matches snapshot 1`] = `
<body>
  <div>
    <div
      class="makeStyles-pageContainer-1"
    >
      <img
        alt="The LCI-lab logo"
        class="makeStyles-pageLogo-2"
        src="LCI_logo.png"
      />
      <div
        class="makeStyles-pageContent-3"
      >
        <div
          class="makeStyles-form-5"
        >
          <h1
            class="MuiTypography-root MuiTypography-h1"
            style="color: rgb(0, 0, 0);"
          >
            Create New Learning Session
          </h1>
          <div
            class="MuiFormControl-root MuiTextField-root makeStyles-inputNewSessionName-6"
          >
            <label
              class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
              data-shrink="false"
            >
              Session name
            </label>
            <div
              class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiInput-input"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class="makeStyles-addStudentContainer-10"
          >
            <div
              class="MuiFormControl-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root makeStyles-placeholder-14 MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
                id="label"
              >
                Student
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <div
                  aria-haspopup="listbox"
                  aria-labelledby="label"
                  class="MuiSelect-root makeStyles-inputSelectStudent-8 MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
                  role="button"
                  tabindex="0"
                >
                  <span>
                    â€‹
                  </span>
                </div>
                <input
                  aria-hidden="true"
                  class="MuiSelect-nativeInput"
                  tabindex="-1"
                  value=""
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSelect-icon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
            </div>
            <button
              aria-label="add new student"
              class="MuiButtonBase-root MuiIconButton-root makeStyles-addStudentBtn-9 MuiIconButton-colorPrimary"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm4 11h-3v3c0 .55-.45 1-1 1s-1-.45-1-1v-3H8c-.55 0-1-.45-1-1s.45-1 1-1h3V8c0-.55.45-1 1-1s1 .45 1 1v3h3c.55 0 1 .45 1 1s-.45 1-1 1z"
                  />
                </svg>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </div>
          <div>
            <span
              class="MuiTypography-root MuiTypography-caption"
            >
              Eye tracking device
            </span>
            <div
              aria-label="eye tracker decive"
              class="MuiFormGroup-root MuiFormGroup-row"
              role="radiogroup"
            >
              <label
                class="MuiFormControlLabel-root"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-18"
                      name="eyetracker"
                      type="radio"
                      value="stationary"
                    />
                    <div
                      class="PrivateRadioButtonIcon-root-19"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-20"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </div>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Stationary
                </span>
              </label>
              <label
                class="MuiFormControlLabel-root"
                style="margin-left: 20px;"
              >
                <span
                  aria-disabled="false"
                  class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-15 MuiRadio-root MuiRadio-colorSecondary MuiIconButton-colorSecondary"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <input
                      class="PrivateSwitchBase-input-18"
                      name="eyetracker"
                      type="radio"
                      value="mobile"
                    />
                    <div
                      class="PrivateRadioButtonIcon-root-19"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                        />
                      </svg>
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-20"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                        />
                      </svg>
                    </div>
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </span>
                <span
                  class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
                >
                  Mobile
                </span>
              </label>
            </div>
          </div>
          <div
            class="makeStyles-sessionInfoContainer-12"
          >
            <span
              class="MuiTypography-root MuiTypography-caption"
            >
              Session code
            </span>
            <span
              class="MuiTypography-root makeStyles-sessionToken-11 MuiTypography-caption"
            />
            <span
              class="MuiTypography-root MuiTypography-caption"
              style="color: rgb(221, 87, 87);"
            >
              Waiting for student connection...
            </span>
          </div>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-13 MuiButton-containedPrimary Mui-disabled Mui-disabled"
            disabled=""
            name="submitBtn"
            tabindex="-1"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Create session
            </span>
          </button>
        </div>
        <img
          alt="dashboard illustration"
          class="makeStyles-illustration-4"
          src="dashboardIllustration.svg"
        />
      </div>
    </div>
    <div
      aria-hidden="true"
      class="MuiBackdrop-root makeStyles-backdrop-22"
      style="opacity: 0; visibility: hidden;"
    >
      <div
        class="MuiPaper-root MuiCard-root makeStyles-card-23 MuiPaper-elevation1 MuiPaper-rounded"
      >
        <div
          class="MuiCardContent-root makeStyles-cardContent-24"
        >
          <div
            class="makeStyles-form-25"
          >
            <h1
              class="MuiTypography-root MuiTypography-h1"
              style="color: rgb(0, 0, 0);"
            >
              Add Student
            </h1>
            <div
              class="MuiFormControl-root MuiTextField-root makeStyles-inputNewStudentName-26"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
                data-shrink="false"
              >
                Student name
              </label>
              <div
                class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiInput-input"
                  data-testid="content-input"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="MuiCardActions-root MuiCardActions-spacing"
              style="width: 100%; justify-content: flex-end;"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  Cancel
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-btn-27 MuiButton-containedPrimary"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiButton-startIcon MuiButton-iconSizeMedium"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
                      />
                    </svg>
                  </span>
                  Add student
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
